src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs

```diff
@@ -187,6 +187,16 @@ private object TransformInput(object input, Dictionary<string, object> bindingDa
             return input;
         }
 
+        private CancellationToken HandleCancellationTokenParameter(object input)
+        {
+            if (input == null)
+            {
+                return CancellationToken.None;
+            }
+
+            return (CancellationToken)input;
```

Comment:
Another silly question: Will the OOP worker actually use this CancellationToken instance from this payload in the invocation request? I thought (for example, in the dotnet worker), it is going to create a token source for each invocation and when the `InvocationCancel` GRPC message arrives, it call Cancel on the token.

Reply:
Nope, it will not use this instance. The OOP worker will create its own CancellationTokenSource and cancel it when it receives the InvocationCancel message (in the dotnet scenario, this will ofc look different for each langauge).

This cancellation token comes from WebJobs and when cancelled (for timeout or other reasons), will lead to the host sending a InvocationCancel message. The workers will not have any ties to this CT

Reply:
@kshyju a quick draft I have been using to test this E2E: https://github.com/Azure/azure-functions-dotnet-worker/pull/972

Feel free to suggest a different approach here, I'm not familiar with the dotnet worker codebase and best practices!